elasticdl/python/data/reader/data_reader.py

```diff
@@ -1,11 +1,33 @@
 from abc import ABC, abstractmethod
 
+from elasticdl.python.common.dtypes import MAXCOMPUTE_DTYPE_TO_TF
+
 
 class Metadata(object):
     def __init__(self, column_names, column_dtypes=None):
         self.column_names = column_names
         self.column_dtypes = column_dtypes
 
+    def get_tf_dtype_by_maxcompute(self, column_name):
+        """Get TensorFlow dtype according to the column name in
+
+        Args:
+            column_name: The column name in a MaxCompute table.
+
+        Returns:
+            TensorFlow dtype
+        """
+        maxcompute_dtype = self.column_dtypes[
+            self.column_names.index(column_name)
```

Comment:
Add a check for `column_dtypes` ? It can be None.

Reply:
`index` is `O(n)`.  You can use a dict, keys are names, values are types? And add `column_names` and `column_types` as a property.

Reply:
Good! I have fixed `column_dtypes` to a dict.